include/GoogleSync/GoogleSync.php

```diff
@@ -182,23 +195,42 @@ public function getClient($id)
     {
         // Retrieve user bean
         if (!isset($this->workingUser)) {
-            $this->workingUser = new User();
+            $this->workingUser = BeanFactory::getBean('Users', $id);
+            if (!$this->workingUser) {
+                throw new Exception('Unable to retrieve a User bean');
```

Comment:
Please add a `@throws Exception` to the phpdoc above the function. 
(Also if you add a unique const code to the exception, the caller can handle the errors more easy depends on what error happend exactly. For e.g:)
```
throw new Exception('Unable to retrieve a User bean', GoogleSync::GSYNC_ERR_NO_USER);
...
...
catch (Exception $e) {
  switch ($e->getCode()) {
    case GoogleSync::GSYNC_ERR_NO_USER:
      echo 'Dear user, I am theribly sorry, but I couldn\'t resolve the requested user for you, please see the log for more info..';
      log('Google Sync error: ' . $e->getMessage());
      break;
    default:
      echo 'Dear user, something went wrong...';
      log('An error: ' . $e->getMessage());
      break;
}
```

(Also please update the phpdoc everywhere where exceptions uses so that third party developers can read your code more easy)

Reply:
I think I did one better than a const code, but let me know what you think.
I extended the Exception class for each type of exception (11 in all). This means the code is baked into the exception, and we can use multiple catch statements to see what exceptions were thrown.
I also added a test for each new exception.